### PR TITLE
EDGE-278 Make Connect SDK use Boost from artifactory

### PR DESCRIPTION
**Pull request by ionyshch-prismsl**
Currently merging my another pull request is blocked by missing/wrong Boost libraries, thus switch to using Boost from artifactory.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,5 @@ connect.files
 connect.includes
 *.tar.gz
 .DS_Store
+ext/boost_macos_1.62.0
+ext/opencv_macos_3.2.0
```
Comment (kyrylo-shpytsya-prism) [inline .gitignore]: How about a "*" instead of a specific version and maybe instead of "macos" as well -- there are linux users, you know.

File: platforms/hanwha-wn/CMakeLists.txt [hunk 2]
```diff
@@ -57,4 +57,3 @@ set (Boost_INCLUDE_DIRS ${CMAKE_INSTALL_PREFIX}/include)
 set (Boost_LIBRARY_DIRS ${CMAKE_INSTALL_PREFIX}/lib)
 
 buildSdk()
-
```
Comment (kyrylo-shpytsya-prism) [inline platforms/hanwha-wn/CMakeLists.txt]: Is this cosmetic fix relevant to this commit?

Comment (ionyshch-prismsl) [reply platforms/hanwha-wn/CMakeLists.txt]: Nope. But as YuriyS once noted, there will be no task for such changes.

File: .gitignore (hunk 1 above)
Comment (ionyshch-prismsl) [reply .gitignore]: Agree. Will make it ext/boost_* ext/opencv_*

File: platforms/shared/shared.cmake [hunk 3]
```diff
@@ -1,26 +1,82 @@
...
+    list (GET status 0 retCode)
+    list (GET status 1 retMessage)
+
+    if (NOT "${retCode}" STREQUAL "0")
+        message(FATAL_ERROR "Error downloading artifact: ${retCode} ${retMessage}")
+        execute_process(COMMAND rm -rf ./* WORKING_DIRECTORY ${dstPath})
     endif()
 
+    execute_process(
+       COMMAND tar -xvzf ${archiveName}
```
Comment (kyrylo-shpytsya-prism) [inline platforms/shared/shared.cmake:31]: Have we considered using cmake's builtins https://cmake.org/cmake/help/v3.2/manual/cmake.1.html#command-line-tool-mode ?

File: platforms/shared/shared.cmake [hunk 4]
```diff
@@ -1,26 +1,82 @@
...
+       WORKING_DIRECTORY ${dstPath}
+       RESULT_VARIABLE resultVar
+       ERROR_VARIABLE errorVar
+    )
+
+    if (NOT "${resultVar}" STREQUAL "0")
+        message(FATAL_ERROR "Error unpacking artifact archive: ${errorVar}")
+        execute_process(COMMAND rm -rf ./* WORKING_DIRECTORY ${dstPath})
```
Comment (kyrylo-shpytsya-prism) [inline platforms/shared/shared.cmake:39]: use builtin?

File: platforms/shared/shared.cmake [hunk 5]
```diff
@@ -1,26 +1,82 @@
...
+        download_artifact("${ARTIFACTORY_URL}/libs-release-public" "${PRISM_PLATFORM}" boost "${BOOST_VERSION}" "${BOOST_ROOT}")
+    else ()
+        set(BOOST_ROOT $ENV{BOOST_ROOT})
+    endif ()
+
+    set(Boost_USE_STATIC_LIBS ON)
+    set(Boost_USE_MULTITHREADED ON)
+    set(Boost_USE_STATIC_RUNTIME ON)
+
+    set(BOOST_COMPONENTS thread chrono system date_time atomic filesystem regex program_options)
```
Comment (kyrylo-shpytsya-prism) [inline platforms/shared/shared.cmake:55]: Maybe move this list to a separate "constant" defined at the top of the file?

Comment (ionyshch-prismsl) [reply platforms/shared/shared.cmake:55]: May be, but my reasoning is either one uses findBoostCommon() with this list, or defines his/her own list of components and doesn't need it.

File: platforms/hanwha-wn/CMakeLists.txt (hunk 2 above)
Comment (kyrylo-shpytsya-prism) [reply platforms/hanwha-wn/CMakeLists.txt]: But as far as I understand, one can always make a trivial pull request consisting solely of cosmetic changes and it should be trivial to have an approval.
I am still strongly opposed to piggybacking of cosmetic fixes, no matter how small, to other commits.

File: platforms/shared/shared.cmake (hunk 3 above)
Comment (ionyshch-prismsl) [reply platforms/shared/shared.cmake:31]: Not sure, that using builtins is better option. Seems we still create process, and with execute_process() there is a very straightforward way to get process return value, output to stdout and stderr.

Comment (kyrylo-shpytsya-prism) [reply platforms/shared/shared.cmake:31]: The question is, do we rely on POSIX environment for the build? I.e. will we ever want to build on Windows™ where "tar" and "rm" are not standard things to have?
My idea was to still use execute_process, but use cmake's portable implementations for "tar" and "rm" ("remove_dirrectory"). See https://cmake.org/cmake/help/v3.2/variable/CMAKE_COMMAND.html

Comment (ionyshch-prismsl) [reply platforms/shared/shared.cmake:31]: I don't know, if we going to support Windows any time soon. I see your point and I think it deserves to be a Jira task to switch to using builtins, where possible.

**Review by kyrylo-shpytsya-prism — APPROVED** [review]
